Sources/TuistSupport/SwiftPackageManager/PackageInfo.swift

```diff
@@ -492,3 +492,107 @@ extension PackageInfo.Product.ProductType: Decodable {
         }
     }
 }
+
+extension PackageInfo.Target.TargetType {
+    /// Defines if the target would be processed when processing the package
+    public var isSupported: Bool {
+        switch self {
+        case .regular, .system:
+            return true
+        default:
+            return false
+        }
+    }
+
+    /// Defines if target may have a public headers path
+    /// Based on preconditions in https://github.com/apple/swift-package-manager/blob/main/Sources/PackageDescription/Target.swift
+    public var supportsPublicHeaderPath: Bool {
+        switch self {
+        case .regular, .executable, .test:
+            return true
+        case .system, .binary, .plugin:
+            return false
+        }
+    }
+
+    /// Defines if target may have source files
+    /// Based on preconditions in https://github.com/apple/swift-package-manager/blob/main/Sources/PackageDescription/Target.swift
+    public var supportsSources: Bool {
+        switch self {
+        case .regular, .executable, .test, .plugin:
+            return true
+        case .system, .binary:
+            return false
+        }
+    }
+
+    /// Defines if target may have resource files
+    /// Based on preconditions in https://github.com/apple/swift-package-manager/blob/main/Sources/PackageDescription/Target.swift
+    public var supportsResources: Bool {
+        switch self {
+        case .regular, .executable, .test:
+            return true
+        case .system, .binary, .plugin:
+            return false
+        }
+    }
+
+    /// Defines if target may have other dependencies
+    /// Based on preconditions in https://github.com/apple/swift-package-manager/blob/main/Sources/PackageDescription/Target.swift
+    public var supportsDependencies: Bool {
+        switch self {
+        case .regular, .executable, .test, .plugin:
+            return true
+        case .system, .binary:
+            return false
+        }
+    }
+
+    /// Defines if target supports C settings
+    /// Based on preconditions in https://github.com/apple/swift-package-manager/blob/main/Sources/PackageDescription/Target.swift
+    public var supportsCSettings: Bool {
```

Comment:
This and similar ones can be removed now that you have `supportsCustomSettings`, isn't it?

Reply:
`supportsCustomSettings` is basically an `or` combination of these variables. Should I remove them or keep them for reference on the origin of these settings?

Reply:
I would remove them and have a single switch in `supportsCustomSettings`

Reply:
ok, removed